File: timeboard/timeboard.rb
require 'rubygems'
require 'dogapi'

api_key=ENV['DD_API_KEY']
app_key=ENV['DD_APP_KEY']

dog = Dogapi::Client.new(api_key, app_key)


# Create a timeboard.
title = 'Scott Ford - Timeboard'
description = 'A basic timeboard generated via the API'
graphs = [{
  "definition" => {
    "events" => [],
    "requests"=> [
      {
        "q" => "avg:my_metric{host:mongodb-ubuntu-1804}",
        "type" => "line"
      }
    ],
  "viz" => "timeseries"
  },
  "title" => "My metric over host"
},
{
  "definition" => {
    "events" => [],
    "requests"=> [
      {
        "q" => "anomalies(avg:mongodb.mem.virtual{*}, 'basic', 2)",
        "type" => "line"
      }
    ],
  "viz" => "timeseries"
  },
  "title" => "MongoDB anomolies"
},
{
"definition" => {
    "events" => [],
    "requests"=> [
      {
        "q" => "avg:my_metric{*}.rollup(sum, 3600)",
        "type" => "line"
      }
    ],
  "viz" => "timeseries"
  },
  "title" => "my_metric rollup sum over one hour"
}]
template_variables = [{
	"name" => "mongodb-ubuntu-1804",
	"prefix" => "host",
	"default" => "host:mongodb-ubuntu-1804"
}]

res = dog.create_dashboard(title, description, graphs, template_variables)

File: timeboard/timeboard.py
from datadog import initialize, api

options = {
    'api_key': '75cc324da0bc265b8883ce646853b814',
    'app_key': '90010fe90c75c6ba58575f40fe0b5831b2db00fd'
}

initialize(**options)

title = 'My API Created Dashboard'
widgets = [{
    'definition': {
        'type': 'timeseries',
        'requests': [
            {'q': 'avg:system.mem.free{*}'}
        ],
        'title': 'Average Memory Free'
    }
}]
layout_type = 'ordered'
description = 'A dashboard with memory info.'
is_read_only = True
notify_list = ['tomolry@optonline.net']
template_variables = [{
    'name': 'datadog1',
    'prefix': 'test',
    'default': 'datadog1'
}]

saved_view = [{
    'name': 'Saved views for hostname 2',
    'template_variables': [{'name': 'host', 'value': 'datadog1'}]}
]

api.Dashboard.create(title=title,
                     widgets=widgets,
                     layout_type=layout_type,
                     description=description,
                     is_read_only=is_read_only,
                     notify_list=notify_list,
                     template_variables=template_variables,
                     template_variable_presets=saved_views)
